File: app/src/main/java/com/cheatdatabase/events/CheatRatingFinishedEvent.java
package com.cheatdatabase.events;

import com.cheatdatabase.model.Cheat;

public class CheatRatingFinishedEvent {

    private Exception mException;
    private boolean mSucceeded;
    private Cheat mCheat;
    private int mRating;

    public CheatRatingFinishedEvent(Cheat cheat, int rating) {
        mSucceeded = true;
        mCheat = cheat;
        mRating = rating;
    }

    public CheatRatingFinishedEvent(Exception exception) {
        mSucceeded = false;
        mException = exception;
    }

    public Cheat getCheat() {
        return mCheat;
    }

    public int getRating() {
        return mRating;
    }

    public Exception getException() {
        return mException;
    }

    public boolean isSucceeded() {
        return mSucceeded;
    }
}

File: app/src/main/java/com/cheatdatabase/events/CheatRatingFinishedEvent.kt
package com.cheatdatabase.events

import com.cheatdatabase.data.model.Cheat

data class CheatRatingFinishedEvent(var isSucceeded: Boolean) {

    lateinit var throwable: Throwable
    lateinit var cheat: Cheat
    var rating: Int = 0

    constructor(cheat: Cheat, rating: Int, isSucceeded: Boolean) : this(isSucceeded) {
        this.isSucceeded = isSucceeded
        this.cheat = cheat
        this.rating = rating
    }

    constructor(throwable: Throwable, isSucceeded: Boolean) : this(isSucceeded) {
        this.isSucceeded = isSucceeded
        this.throwable = throwable
    }
}